### ISSUE TITLE
CMake FetchContent not working as expected

### ISSUE THREAD

**Issue by OlivierLDff** [google/googletest#2457]
## Building gtest as part of a CMake project fails (sometimes)

I'm facing some issue building gtest inside my project using FetchContent. I don't know if problem is coming from gtest or from cmake.

### Environnement
* Windows 10 (seems to work in other environment)
* CMake generator : Visual Studio 16 2019
* MSVC 19.22.27905.0
* Tested with CMake 3.13 and 3.15.
* gtest version master and 1.8.0 tested

And my test command with the following code snippets.
```
mkdir build && cd build
cmake ..
cmake --build .
```

### Working example

So here a basic example working to integrate gtest inside a project that is working perfectly fine.

```cmake
cmake_minimum_required(VERSION 3.11)

project("gTestDemoProject")

include(FetchContent)

FetchContent_Declare(
    googletest
    GIT_REPOSITORY "https://github.com/google/googletest"
    GIT_TAG        master
)

set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
FetchContent_MakeAvailable(googletest)
```

### Not working example

And here is the code snippet not working.

```cmake
cmake_minimum_required(VERSION 3.13)

project("gTestDemoProject")

include(FetchContent)

FetchContent_Declare(
    gtest
    GIT_REPOSITORY "https://github.com/google/googletest"
    GIT_TAG        master
)

set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
FetchContent_MakeAvailable(gtest)
```
Only value changing is calling the content gtest instead of googletest.
If you try to compile gmock will fail because it can't find gtest include folder. gtest still build nicely. 

After investigating it seems that `gtest_SOURCE_DIR` value is wrong inside `googlemock/CMakeLists.txt`. The value is `path/to/repository` instead of `path/to/repository/googletest`. 

If I print the value of  `gtest_SOURCE_DIR` inside `googletest/CMakeLists.txt` it will be `path/to/repository/googletest` (so normal value). But once the `add_subdirectory` in `googlemock/CMakeLists.txt` return the value become `path/to/repository`.

So i would like to know if someone is able to reproduce this bug or if this is coming from my configuration (I only have access to one windows machine for the time). Is this a bug from CMake/FetchContent function?

Cheers.
Olivier.

**Comment by ncatta** [comment]
Hello,

I reproduced this on Ubuntu 18.04, CMake 3.15 and gcc 7.4.0.

**Comment by yamavol** [comment]
Try this. 
```
FetchContent_Declare(
    googletest
    GIT_REPOSITORY https://github.com/google/googletest.git
    GIT_TAG        release-1.10.0
    BUILD_COMMAND
)
FetchContent_MakeAvailable(googletest)
target_link_libraries(app gtest) 
target_include_directories(app PRIVATE ${googletest_SOURCE_DIR}/googletest/include/)
```
FetchContent only downloads and populate it; it won't automatically build the library.
ref: http://cmake.3232098.n2.nabble.com/How-to-link-against-projects-added-through-FetchContent-td7597224.html

**Comment by gennadiycivil** [comment]
Anything related to CMake is community supported. We do not use it internally and we are not in position to check. In general the maintainers would look for community consensus.

**Comment by alex1989hu** [comment]
I cannot reproduce it neither on Ubuntu 18.04.3 LTS, nor on Scientific Linux 7.6. CMake versions were: 3.14.5 and 3.15.4. I tried several times without any issues.

*PS. I used your gtest snippet to get set gtest as FetchContent_Declare name.*

**Comment by OlivierLDff** [comment]
Ok issue is fixed for me too. I tried with commit `3a45039862471cc2785e92bd19bd146c70344986` that was master when i submitted this issue.
* MSVC 19.23.28106.4 (so that updated).
* cmake 3.15.3 (The version i was using when submitting the issue).
* Windows 1903 - 18362.418 (that have been updated)

So it's a very strange issue because ncatta reproduced it on ubuntu with gcc so we it's not windows update fault nor msvc. I think we can close the issue but it remains very strange.
